ModuleNotFoundError with pytest-asyncio

Thanks for providing this plugin, it looks really useful.

I'm trying to use it in an async project and getting an error which I've pinned down to the combination of this plugin and `pytest-asyncio`. I admittedly don't know which plugin might need to change here, though given the popularity of async code it would be great if the two worked together.

### Reproduce

`README.md`:
<pre>
``` python
def test():
    pass

```
``` pycon
>>> 1 + 1
2

```
</pre>

``` console
$ pip install pytest-asyncio pytest-doctestplus
...
$ pytest --doctest-plus --doctest-glob '*.md'
=========================== test session starts ============================
platform linux -- Python 3.10.12, pytest-8.2.2, pluggy-1.5.0
rootdir: /tmp/bees
plugins: doctestplus-1.2.1, asyncio-0.23.7
asyncio: mode=strict
collected 0 items / 1 error                                                

================================== ERRORS ==================================
________________________ ERROR collecting README.md ________________________
ImportError while importing test module '/tmp/bees/README.md'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ModuleNotFoundError: No module named 'README'
========================= short test summary info ==========================
ERROR README.md
!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!
============================= 1 error in 0.20s =============================
```

--- a/.github/workflows/python-tests.yml
+++ b/.github/workflows/python-tests.yml
@@ -70,6 +70,9 @@ jobs:
           - os: ubuntu-latest
             python-version: '3.13'
             toxenv: py313-test-pytestdev-numpydev
+          - os: ubuntu-latest
+            python-version: '3.13'
+            toxenv: py313-test-pytest83-pytestasyncio
 
     steps:
     - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,6 +1,8 @@
 1.4.0 (unreleased)
 ==================
 
+- Fixing compatibility with pytest-asyncio. [#278]
+
 - Versions of Python <3.9 are no longer supported. [#274]
 
 1.3.0 (2024-11-25)

--- a/pytest_doctestplus/plugin.py
+++ b/pytest_doctestplus/plugin.py
@@ -342,6 +342,7 @@ def pytest_configure(config):
                             test.name, self, runner, test)
 
     class DocTestTextfilePlus(pytest.Module):
+        obj = None
 
         def collect(self):
             if PYTEST_GE_7_0:

--- a/tests/test_doctestplus.py
+++ b/tests/test_doctestplus.py
@@ -11,6 +11,13 @@ import pytest
 import doctest
 from pytest_doctestplus.output_checker import OutputChecker, FLOAT_CMP
 
+try:
+    import pytest_asyncio  # noqa: F401
+    has_pytest_asyncio = True
+except ImportError:
+    has_pytest_asyncio = False
+
+
 
 pytest_plugins = ['pytester']
 
@@ -1123,6 +1130,9 @@ def test_fail_data_dependency(testdir, cont_on_fail):
     assert ("something()\nUNEXPECTED EXCEPTION: NameError" in report.longreprtext) is cont_on_fail
 
 
+@pytest.mark.xfail(
+        has_pytest_asyncio,
+        reason='pytest_asyncio monkey-patches .collect()')
 def test_main(testdir):
     pkg = testdir.mkdir('pkg')
     code = dedent(

--- a/tox.ini
+++ b/tox.ini
@@ -32,6 +32,7 @@ deps =
     pytest83: pytest==8.3.*
     pytestdev: git+https://github.com/pytest-dev/pytest#egg=pytest
     numpydev: numpy>=0.0.dev0
+    pytestasyncio: pytest-asyncio
 
 extras =
     test

